Library/Formula/mysql.rb

```diff
@@ -53,18 +53,18 @@ def install
     # compilation of gems and other software that queries `mysql-config`.
     ENV.minimal_optimization
 
-    # -DINSTALL_* are relative to prefix
+    # -DINSTALL_* are relative to `CMAKE_INSTALL_PREFIX` (`prefix`)
     args = %W[
       .
       -DCMAKE_INSTALL_PREFIX=#{prefix}
       -DCMAKE_FIND_FRAMEWORK=LAST
       -DCMAKE_VERBOSE_MAKEFILE=ON
       -DMYSQL_DATADIR=#{datadir}
-      -DINSTALL_INCLUDEDIR=include/mysql
-      -DINSTALL_MANDIR=share/man
-      -DINSTALL_DOCDIR=share/doc/#{name}
-      -DINSTALL_INFODIR=share/info
-      -DINSTALL_MYSQLSHAREDIR=share/mysql
+      -DINSTALL_INCLUDEDIR=#{include.relative_path_from(prefix)}/mysql
+      -DINSTALL_MANDIR=#{man.relative_path_from(prefix)}
+      -DINSTALL_DOCDIR=#{doc.relative_path_from(prefix)}
+      -DINSTALL_INFODIR=#{info.relative_path_from(prefix)}
+      -DINSTALL_MYSQLSHAREDIR=#{share.relative_path_from(prefix)}/mysql
```

Comment:
Ultimately yes, but if the method `share` changes to return something other than `prefix+'share'`, this will continue to return the correct value, while the old code would not